File: docs/get_show_data.js
function get_show_data(user_list) {
  var shows = {};
  var i = 0;

  for (i; i < user_list.length; i++) {
    const show_data = query_name(user_list[i]['title'], user_list[i]['id']);
    if (show_data) {
      shows[i] = show_data['search'];
      console.log(show_data['search']['anime'][0]['name']);
      i++;
    }
  }

  return shows;
}

function query_name(name, id) {
  const url = 'https://api.animethemes.moe/search'

  const params = {
    'q': [
      name
    ],

    'page[limit]': [
      1
    ],

    'include[anime]': [
      'animethemes.song.artists',
      'animethemes.animethemeentries.videos',
      'resources'
    ],

    'fields[search]': [
      'anime'
    ],

    'fields[anime]': [
      'year',
      'name'
    ],

    'fields[song]': [
      'title'
    ],

    'fields[artist]': [
      'name'
    ],

    'fields[video]': [
      'resolution',
      'link'
    ],

    'fields[animethemeentry': [
      'id'
    ],

    'fields[resource]': [
      'external_id',
      'site'
    ]
  }

  const options = {
    method: 'GET',

  }

  const show_data = fetch(url + '?' + new URLSearchParams(params)).json();

  if (show_data['search']['anime'][0]['resources'][0]['site'] === 'MyAnimeList') {
    const mal_id = show_data['search']['anime'][0]['resources'][0]['external_id'];
  } else {
    const mal_id = 0;
  }

  if (mal_id === id) return show_data;

  return;
}

File: docs/python/get_show_data.py
import requests

def get_show_data(user_list):
  ids = ""
  ids+=user_list[0]["id"]

  for entry in range(len(user_list)-1):
    ids+=","+user_list[entry+1]["id"]
  shows = query_name(ids) 

  return shows
  # with open('lists/show_data.json', 'w') as outfile:
  #   json.dump(shows, outfile, indent=2)

def query_name(ids):
  list_data = {"anime": []}

  url = f'https://api.animethemes.moe/anime'
  
  params = {

    "include":
      "animethemes.song.artists,animethemes.animethemeentries.videos,resources",
    
    "fields[search]":
      "anime",

    "fields[anime]":
      "year,name",

    "fields[song]":
      "title",

    "fields[artist]" :
      "name",

    "fields[animetheme]":
      "type,sequence",

    "fields[video]":
      "resolution,link",

    "fields[animethemeentry]":
      "id",

    "fields[resource]":
      "external_id,site",

    "filter[external_id]":
      ids,

    "filter[site]":
      "MyAnimeList",

    "page[number]":
      1
  }



  return requests.get(url, params=params).json()